src/crimson/os/seastore/transaction_manager.h

```diff
@@ -322,6 +322,28 @@ class TransactionManager : public SegmentCleaner::ExtentCallbackInterface {
     return segment_manager.release(id);
   }
 
+  /**
+   * get_root
+   *
+   * Get root block's ondisk layout
+   */
+  using get_root_ertr = base_ertr;
+  using get_root_ret = get_root_ertr::future<RootBlockRef>;
+  get_root_ret get_root(Transaction &t) {
+    return cache->get_root(t);
+  }
+
+  void update_root_meta(Transaction& t,
+                        const std::string& key,
+                        const std::string& value) {
+    auto root = cache->get_root_fast(t);
+    root = cache->duplicate_for_write(t, root)->cast<RootBlock>();
+    root->meta[key] = value;
+    ceph::bufferlist bl(1);
+    bl.rebuild(ceph::buffer::ptr_node::create(&root->get_root().meta[0], root_t::MAX_META_LENGTH));
```

Comment:
might want to error out if the total size of meta map exceeds the capacity of `meta[]`? also, i think we can save the space using `string_view` here.